ckanext/scheming/templates/scheming/package/resource_read.html

```diff
@@ -9,58 +9,61 @@
-            <td>{{ h.render_datetime(res.created) or _('unknown') }}</td>
-          </tr>
-        {%- endblock -%}
-        {%- block resource_format -%}
-          <tr>
-            <th scope="row">{{ _('Format') }}</th>
-            <td>{{ res.mimetype_inner or res.mimetype or res.format or _('unknown') }}</td>
-          </tr>
-        {%- endblock -%}
-        {%- block resource_license -%}
-          <tr>
-            <th scope="row">{{ _('License') }}</th>
-            <td>{% snippet "snippets/license.html", pkg_dict=pkg, text_only=True %}</td>
-          </tr>
-        {%- endblock -%}
-        {%- block resource_fields -%}
-          {%- for field in schema.resource_fields -%}
-            {%- if field.field_name not in exclude_fields
-                and field.display_snippet is not none -%}
-              <tr>
-                <th scope="row">
-                  {{- h.scheming_language_text(field.label) -}}
-                </th>
-                <td>
-                  {%- snippet 'scheming/snippets/display_field.html',
-                      field=field, data=res, entity_type='dataset',
-                      object_type=dataset_type -%}
-                </td>
-              </tr>
-            {%- endif -%}
-          {%- endfor -%}
-        {%- endblock -%}
-      </tbody>
-    </table>
-  </div>
+    <div class="module-content">
+        <h2>{{ _('Additional Information') }}</h2>
+        <table class="table table-striped table-bordered table-condensed" data-module="table-toggle-more">
+            <tbody>
+            {%- block resource_last_updated -%}
+                <tr>
+                    <th scope="row">{{ _('Last updated') }}</th>
+                    <td>{{ h.render_datetime(res.last_modified) or h.render_datetime(res.revision_timestamp) or h.render_datetime(res.created) or _('unknown') }}</td>
+                </tr>
+            {%- endblock -%}
+            {%- block resource_created -%}
+                <tr>
+                    <th scope="row">{{ _('Created') }}</th>
+                    <td>{{ h.render_datetime(res.created) or _('unknown') }}</td>
+                </tr>
+            {%- endblock -%}
+            {%- block resource_format -%}
+                <tr>
+                    <th scope="row">{{ _('Format') }}</th>
+                    <td>{{ res.mimetype_inner or res.mimetype or res.format or _('unknown') }}</td>
+                </tr>
+            {%- endblock -%}
+            {%- block resource_license -%}
+                <tr>
+                    <th scope="row">{{ _('License') }}</th>
+                    <td>{% snippet "snippets/license.html", pkg_dict=pkg, text_only=True %}</td>
+                </tr>
+            {%- endblock -%}
+            </tbody>
+        </table>
+
+        {% block resource_fields %}
+            {%- for group in schema.resource_fields|groupby('display_group') -%}
+                {%- set group_fields = h.non_empty_fields(group.list|list, res, exclude_fields) -%}
```

Comment:
Where's this defined?

Reply:
```
def non_empty_fields(field_list, pkg_dict, exclude):
    r = []
    for field in field_list:
        if field['field_name'] in exclude:
            continue

        if field.get('display_snippet', False) is None:
            continue

        if pkg_dict.get(field['field_name']):
            r.append(field)
return r
```

Was missed when pulling chunks out for the PR.